### PR TITLE
Avoid using Invoke-Expression

### PR DESCRIPTION
**Pull request by xtqqczze**
# PR Summary

Remove the use of Invoke-Expression in a few cases as per AvoidUsingInvokeExpression PSScriptAnalyzer rule.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively

### CODE REVIEW

**Comment by adityapatwardhan** [comment]
Closing and re-opening to re-trigger CI

File: build.psm1 [hunk 1]
```diff
@@ -1513,10 +1512,8 @@ function Install-Dotnet {
         }
 
         if ($uninstallScript) {
-            Start-NativeExecution {
-                curl -sO $obtainUrl/uninstall/$uninstallScript
-                Invoke-Expression "$sudo bash ./$uninstallScript"
-            }
+            Start-NativeExecution [scriptblock]::Create("curl -sO $obtainUrl/uninstall/$uninstallScript")
```
Comment (TravisEz13) [inline build.psm1]: `[scriptblock]::Create` and then invoking it is equivalent of Invoke-Expression.

Comment (TravisEz13) [reply build.psm1]:
```suggestion	
            $uninstallScriptUrl = "$obtainUrl/uninstall/$uninstallScript"
            Start-NativeExecution {curl -sO $uninstallScriptUrl}
```

File: build.psm1 [hunk 2]
```diff
@@ -1513,10 +1512,8 @@ function Install-Dotnet {
...
+            Start-NativeExecution [scriptblock]::Create("$sudo bash ./$uninstallScript")
```
Comment (TravisEz13) [inline build.psm1]: Can you make the equivalent fix to what I suggested, here?

File: build.psm1 [hunk 3]
```diff
@@ -1606,10 +1602,8 @@ function Start-PSBootstrap {
                 $originalDebianFrontEnd=$env:DEBIAN_FRONTEND
                 $env:DEBIAN_FRONTEND='noninteractive'
                 try {
-                    Start-NativeExecution {
-                        Invoke-Expression "$sudo apt-get update -qq"
-                        Invoke-Expression "$sudo apt-get install -y -qq $Deps"
-                    }
+                    Start-NativeExecution [scriptblock]::Create("$sudo apt-get update -qq")
```
Comment (TravisEz13) [inline build.psm1]:
```suggestion
                    Start-NativeExecution {$sudo apt-get update -qq}
```

File: build.psm1 [hunk 4]
```diff
@@ -1606,10 +1602,8 @@ function Start-PSBootstrap {
...
+                    Start-NativeExecution [scriptblock]::Create("$sudo apt-get install -y -qq $Deps")
```
Comment (TravisEz13) [inline build.psm1]:
```suggestion
                    Start-NativeExecution {$sudo apt-get install -y -qq $Deps}
```

File: src/PowerShell.Core.Instrumentation/RegisterManifest.ps1 [hunk 5]
```diff
@@ -87,11 +87,11 @@ foreach ($file in $files)
 # Unregister if present. Avoids warnings when registering the manifest
 # and it is already registered.
 Write-Verbose "unregister the manifest, if present: $command"
-Start-NativeExecution {Invoke-Expression $command} $true
+Start-NativeExecution [scriptblock]::Create($command) $true
```
Comment (TravisEz13) [inline src/PowerShell.Core.Instrumentation/RegisterManifest.ps1]: just no

**Comment by TravisEz13** [comment]
FYI, Feature tests are always run.  `[feature]` doesn't doe anything anymore.

File: build.psm1 [hunk 6]
```diff
@@ -1513,10 +1512,8 @@ function Install-Dotnet {
         }
 
         if ($uninstallScript) {
-            Start-NativeExecution {
-                curl -sO $obtainUrl/uninstall/$uninstallScript
-                Invoke-Expression "$sudo bash ./$uninstallScript"
-            }
+            Start-NativeExecution ([scriptblock]::Create("curl -sO $obtainUrl/uninstall/$uninstallScript"))
```
Comment (TravisEz13) [inline build.psm1]: Using create ScriptBlock is worse that just using a script block.  Please change the curl part back.

File: build.psm1 [hunk 7]
```diff
@@ -1513,10 +1512,8 @@ function Install-Dotnet {
...
+            Start-NativeExecution ([scriptblock]::Create("$sudo bash ./$uninstallScript"))
```
Comment (TravisEz13) [inline build.psm1]: To be 100% clear, this is what invoke-expression does... so we are not fixing any issue here.

**Review by TravisEz13 — CHANGES_REQUESTED** [review]
see comment

**Pull request closed without merging**